src/html-imports.js

```diff
@@ -283,8 +284,19 @@
         (document.createElement('template'));
       template.innerHTML = resource;
       if (template.content) {
-        // This creates issues in Safari10 when used with shadydom (see #12).
         content = template.content;
+        // Clone scripts inside templates since they won't execute when the
+        // hosting template is cloned.
+        forEach(content.querySelectorAll('template'), template => {
```

Comment:
Shouldn't we traverse recursively (templates inside templates)?

Reply:
Good point! let's go with your PR :)